src/connector/src/sink/snowflake.rs

```diff
@@ -236,31 +227,87 @@ impl SnowflakeSinkWriter {
                 TimestamptzHandlingMode::UtcString,
                 TimeHandlingMode::String,
             ),
-            row_counter: 0,
-            payload: String::new(),
-            max_batch_row_num,
             // initial value of `epoch` will start from 0
             epoch: 0,
-        }
+            // will be initialized after the begin of each epoch
+            streaming_uploader: None,
+            file_suffix: None,
+            has_data: false,
+        })
     }
 
-    /// reset the `payload` and `row_counter`.
-    /// shall *only* be called after a successful sink to s3.
-    fn reset(&mut self) {
-        self.payload.clear();
-        self.row_counter = 0;
+    /// update the streaming uploader as well as the file suffix.
+    /// note: should *only* be called when a new epoch begins.
+    async fn update_streaming_uploader(&mut self) -> Result<()> {
```

Comment:
How about creating a new `streaming_uploader` every time? Rust encourages using immutable objects.

Reply:
we're creating `streaming_uploader` every time - the mutable reference here is indicating we changing the internal `file_suffix` and `streaming_uploader` of the current `SnowflakeSinkWriter`.